qiskit_experiments/library/randomized_benchmarking/layer_fidelity.py

```diff
@@ -0,0 +1,473 @@
+            for op in self.backend.target.operations:
+                if isinstance(op, Gate) and op.num_qubits == 2:
+                    two_qubit_gate = op.name
+                    LOG.info("%s is set for two_qubit_gate", op.name)
+                    break
+            if not two_qubit_gate:
+                raise QiskitError("two_qubit_gate is not provided and failed to set from backend.")
+        else:
+            if self.backend is None and two_qubit_gate not in GATE_NAME_MAP:
+                raise QiskitError(f"Unknown two_qubit_gate: {two_qubit_gate}.")
+
+        if one_qubit_basis_gates is None:
+            if self.backend is None:
+                raise QiskitError("one_qubit_basis_gates or backend must be supplied.")
+            # Try to set default one_qubit_basis_gates from backend
+            one_qubit_basis_gates = []
+            for op in self.backend.target.operations:
+                if isinstance(op, Gate) and op.num_qubits == 1:
+                    one_qubit_basis_gates.append(op.name)
+            LOG.info("%s is set for one_qubit_basis_gates", str(one_qubit_basis_gates))
+            if not one_qubit_basis_gates:
+                raise QiskitError(
+                    "one_qubit_basis_gates is not provided and failed to set from backend."
+                )
+        else:
+            if self.backend is None:
+                for gate in one_qubit_basis_gates:
+                    if gate not in GATE_NAME_MAP:
+                        raise QiskitError(f"Unknown gate in one_qubit_basis_gates: {gate}.")
+
+        # Set configurable options
+        self.set_experiment_options(
+            lengths=sorted(lengths),
+            num_samples=num_samples,
+            seed=seed,
+            two_qubit_layers=two_qubit_layers,
+            two_qubit_gate=two_qubit_gate,
+            one_qubit_basis_gates=tuple(one_qubit_basis_gates),
+        )
+
+        # Verify two_qubit_gate and one_qubit_basis_gates
+        self.__validate_basis_gates()
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        """Default experiment options.
+
+        Experiment Options:
+            two_qubit_layers (List[List[Tuple[int, int]]]): List of pairs of qubits to run on.
+            lengths (List[int]): A list of layer lengths.
+            num_samples (int): Number of samples to generate for each layer length.
+            seed (None or int or SeedSequence or BitGenerator or Generator): A seed
+                used to initialize ``numpy.random.default_rng`` when generating circuits.
+                The ``default_rng`` will be initialized with this seed value every time
+                :meth:`circuits` is called.
+            two_qubit_gate (str): Two-qubit gate name (e.g. "cx", "cz", "ecr")
+                of which the two qubit layers consist.
+            one_qubit_basis_gates (Tuple[str]): One-qubit gates to use for implementing 1q Cliffords.
```

Comment:
If `one_qubit_basis_gates` isn't specified, IBM backends usually provide `id` as one of the gates, whereas all the LF examples and tests that specify `one_qubit_basis_gates` don't list `id` in the one-qubit gates. Should this be regularized?

Reply:
I don't think we need to care `id` gate so much. `id` is a legacy one-qubit gate (used before `delay` instruction is supported), which should be removed from basis_gates in the future.